src/object/cli_obj.c

```diff
@@ -306,23 +308,15 @@ obj_get_replicas(struct dc_object *obj)
 		return 1;
 
 	if (oc_attr->u.repl.r_num == DAOS_OBJ_REPL_MAX)
-		return obj->cob_shards_nr;
+		return obj->cob_grp_size;
 
 	return oc_attr->u.repl.r_num;
 }
 
 int
 obj_get_grp_size(struct dc_object *obj)
 {
-	struct daos_oclass_attr *oc_attr;
-	unsigned int grp_size;
-
-	oc_attr = daos_oclass_attr_find(obj->cob_md.omd_id);
-	D_ASSERT(oc_attr != NULL);
-	grp_size = daos_oclass_grp_size(oc_attr);
-	if (grp_size == DAOS_OBJ_REPL_MAX)
-		grp_size = obj->cob_shards_nr;
-	return grp_size;
+	return obj->cob_grp_size;
```

Comment:
This function can be removed.

Reply:
right, some other detailed code also can be cleanup in following patch.